lib/_http_client.js

```diff
@@ -82,7 +82,8 @@ function ClientRequest(options, cb) {
     if (!options.hostname) {
       throw new Error('Unable to determine the domain name');
     }
-  } else if (options instanceof url.URL) {
+  } else if (options && options[searchParamsSymbol] &&
+             options[searchParamsSymbol][searchParamsSymbol]) {
```

Comment:
I am now wondering if this way of checking URLs is a bit hard to understand for someone who don't know the implementation details of URL...